app/main.py

```diff
@@ -1,6 +1,23 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cached_results_storage = {}
```

Comment:
make a shorter variable name. Remove `storage` from the variable name won't make it less descriptive

Reply:
Thanks for advice with return statement.... somehow I didn't realize I can do it this way. Fixed now